docs/docs/data-structures/tree/schema-evolution/index.mdx

```diff
@@ -0,0 +1,219 @@
+In practice, the older version of the application must be able to "view" the new document after it has been upgraded.
+
+If a change is made that is backwards compatible, but not forwards compatible, then old versions of the application will "break" when trying to load newer/upgraded documents.
+To prevent this, the application can employ a [staged rollout](./index.mdx#staged-rollouts).
+
+### Schema Change Examples
+
+#### 😎 Chill
+| Backwards Compatible | Forwards Compatible | Rollout Process |
+|-----------------------|----------------------|-----------------|
+| ✅                    | ✅                   | Normal          |
+
+**Examples:**
+- Adding an optional field when compatibility flag is enabled
+
+#### 🌶️ Spicy
+| Backwards Compatible | Forwards Compatible | Rollout Process |
+|-----------------------|----------------------|-----------------|
+| ✅                    | ❌                   | ⚠️ Staged       |
+
+**Examples:**
+- Adding a new allowed type (TODO link to how to doc)
+
+#### 👮 Go to Jail
+| Backwards Compatible | Forwards Compatible | Rollout Process |
+|-----------------------|----------------------|-----------------|
+| ❌                    | ❌                   | 🚫 Forbidden    |
+
+**Examples:**
+- Adding/removing a required field
+- Removing an allowed type from a field
+- Renaming a node identifier
+
+
+### Checking Compatibility
+
+Use [`TreeView.compatibility`](../../../api/fluid-framework/treeview-interface#compatibility-property) to check if your schema changes are compatible with existing documents:
+
+```typescript
+const view = tree.viewWith(configuration);
+const { compatibility } = view;
+
+if (!compatibility.isEquivalent) {
+    if (compatibility.canUpgrade) {
+        view.upgradeSchema(); // Safe to upgrade
+    } else {
+        // Handle incompatible schema
+        throw new Error("Cannot open document with current schema");
+    }
+}
+```
+
+### Monitoring Remote Schema Changes
+
+In collaborative applications, other clients may upgrade the document's schema while you're working.
+Listen for the `schemaChanged` event to handle these remote upgrades:
+
+```typescript
+// Monitor for remote schema changes
+view.events.on("schemaChanged", () => {
+    if (!view.compatibility.canView) {
+        // Document was upgraded to an incompatible schema
+        // Handle gracefully - perhaps reload or notify user
+        throw new Error("Document was upgraded and is no longer compatible");
+    }
+
+    // Schema changed but is still compatible
+    // You may need to refresh UI or handle new data structures
+    console.log("Document schema was updated by another client");
+});
+```
+
+## Schema Upgrade Process
+
+When you make [backwards compatible](./index.mdx#understanding-compatibility) changes to your schema, you can upgrade existing documents to use the new schema using [`view.upgradeSchema()`](../../../api/fluid-framework/treeview-interface#upgradeschema-method):
```

Comment:
This makes it sound like we're introducing `upgradeSchema()` for the first time, but we already showed it in the example above. That's probably okay if we don't expect users to read this doc from top to bottom. But I do think that they need to have read the example above before they read this one - understanding the compatibility API is a prerequisite to understanding the upgrade API.

Reply:
That's probably what I would have done too 👍